### ISSUE TITLE
[apex] UnusedMethod reports methods unused when param List<Sobject> is passed List<SpecificObject__c>

### ISSUE THREAD

**Issue by nwcm** [pmd/pmd#4812]
**Affects PMD Version:** 7.0.0-rc4

**Rule:** [UnusedMethod](https://docs.pmd-code.org/latest/pmd_rules_apex_design.html#unusedmethod)

**Description:**

Rule flags methods with parameters of type `List<SObject>` as unused when never passed that specific generic type.

**Code Sample demonstrating the issue:**

```java
public inherited sharing class SobjectUtilities{
  public static void doWork(){
    List<Account> accounts = new List<Account>();
    SobjectUtilities.someMethod(accounts);
  }

  public static void someMethod(List<SObject> records) {
    return;
  }
}
```

**Expected outcome:**

This method should not be thrown as unused

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *CLI*
